src/assets/TokenRatesController.ts

```diff
@@ -153,14 +224,68 @@ export class TokenRatesController extends BaseController<
     throw new Error('Property only used for setting');
   }
 
+  private async fetchChainSlug(chainId: string): Promise<string | null> {
+    const platforms: [
+      { id: string; chain_identifier: number | null },
+    ] = await handleFetch(COINGECKO_API.getPlatformsURL());
+    const chain = platforms.find(
+      ({ chain_identifier }) =>
+        chain_identifier !== null && String(chain_identifier) === chainId,
+    );
+    return chain?.id || null;
+  }
+
   /**
    * Fetches a pairs of token address and native currency
    *
+   * @param chainSlug - Chain string identifier
    * @param query - Query according to tokens in tokenList and native currency
    * @returns - Promise resolving to exchange rates for given pairs
    */
-  async fetchExchangeRate(query: string): Promise<CoinGeckoResponse> {
-    return handleFetch(this.getPricingURL(query));
+  async fetchExchangeRate(
+    chainSlug: string,
+    query: string,
+  ): Promise<CoinGeckoResponse> {
+    return handleFetch(COINGECKO_API.getTokenPriceURL(chainSlug, query));
+  }
+
+  async getChainSlug(): Promise<string | null> {
+    const { threshold, chainId } = this.config;
+    const { supportedChains } = this.state;
+
+    const chainCache = supportedChains[chainId];
+    // supportedChain has not been created, we skip this execution
+    if (!chainCache) {
```

Comment:
This branch seems like it should be impossible to reach :thinking: Since `supportedChains[chainId]` is set immediately after switching networks.

Reply:
The race condition I was thinking of is:
Given this state:
- `chainCache.slug` is `null`
- `tokens` is a list of tokens

Since we have now two listeners (on assets change and on network change), could it be the case this method executes when the assets change (because of a network change) instead of the network change itself?

There is an open question on the technical proposal about this

Reply:
The thing is, if the assets change before the chain ID does, this would still be referencing the old chain ID. I don't believe it's possible for this to have a new chain ID without the setter for chainId running first.

Reply:
I see it now, this check is unnecessary 👍